src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -21,10 +25,17 @@ internal class ManagedIdentityClient
         private const string WindowsHimdsFilePath = "%Programfiles%\\AzureConnectedMachineAgent\\himds.exe";
         private const string LinuxHimdsFilePath = "/opt/azcmagent/bin/himds";
         internal static ManagedIdentitySource s_sourceName = ManagedIdentitySource.None;
+        internal static ITimeService s_timeService = new TimeService();
+        internal static readonly TimeSpan MtlsCertRefreshSkew = TimeSpan.FromMinutes(5);
+
+        internal static readonly ConcurrentDictionary<string,
+            (X509Certificate2 Cert, string ClientId, string TenantId, string Endpoint)> s_miCerts = new();
 
         internal static void ResetSourceForTest()
         {
             s_sourceName = ManagedIdentitySource.None;
+            s_miCerts.Clear();
+            s_timeService = new TimeService();
```

Comment:
The point of the time service is that it should be a singleton. A single instance is injected in all classes that deal with time. That instance can either be the real time or a test time. Instead, you are using many instances, so you are not really sure what you test.

Reply:
But I agree that a big refactor here could take time, maybe add a task to use TimeService everywhere in MSAL where DateTime.now is used.